src/Storages/PostgreSQL/PostgreSQLReplicationHandler.cpp

```diff
@@ -224,10 +227,19 @@ void PostgreSQLReplicationHandler::startSynchronization(bool throw_on_error)
 }
 
 
-StoragePtr PostgreSQLReplicationHandler::loadFromSnapshot(String & snapshot_name, const String & table_name,
+ASTPtr PostgreSQLReplicationHandler::getCreateNestedTableQuery(StorageMaterializedPostgreSQL * storage, const String & table_name)
+{
+    postgres::Connection connection(connection_info);
```

Comment:
I see pqlib (and pqxx) for the first time. All connections will be correctly closed in destructors?

Reply:
BTW, is it ok to create new connection each time?

Reply:
> I see pqlib (and pqxx) for the first time. All connections will be correctly closed in destructors?

yes, they will be closed.

> BTW, is it ok to create new connection each time?

Yes, quite bad.
I cannot make one global connection for all methods in this replication handler becuase it is dangerous as they can be called concurrently.
A solution could be to add a connection pool here. But I am not sure it is for the better because all these connections are opened only while initial tables dump or when some non ordinary rare actions are made.
Overall in normal lifestyle of this engine (after initial synchronization is completed) there will be only 1 active connection at a time.